### PR TITLE
feat(nav-drawer): upgrade to v6 and enable freeze

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Updates @react-navigation/drawer to v6 and enables freeze for performance improvements. Some prop changes were needed to maintain existing styles and behavior.

Freeze works as follows:
> ...Using Suspense mechanism introduced in React 17. The main use case of this library is to avoid unnecessary re-renders of parts of the app that are not visible to the user at a given moment. The frozen components are not unmounted when they are replaced with a placeholder view, so their React state and corresponding native view instances are retained during the freeze (DOM elements for react-dom or platform-native views for React Native apps) keeping things like scroll position, input state, or loaded images (for  components) unchanged. [source](https://github.com/software-mansion/react-freeze).

| iOS Before | iOS After | Android Before | Android After |
| :---: | :---: | :---: | :---: |
|  | |  |  |

### Test plan

- Tested locally on Android & iOS

### Related issues

N/A

### Backwards compatibility

Yes

### CODE REVIEW

File: src/qrcode/QRScanner.tsx [hunk 1]
```diff
@@ -69,6 +69,7 @@ export default function QRScanner({ onBarCodeDetected }: QRScannerProps) {
   }
 
   const submitModal = () => {
+    Keyboard.dismiss()
```
Comment (dievazqu) [inline src/qrcode/QRScanner.tsx:72]: Is this change related to the freeze feature?

File: src/navigator/DrawerNavigator.tsx [hunk 2]
```diff
@@ -149,15 +150,19 @@ function CustomDrawerItemList({
         label={drawerLabel !== undefined ? drawerLabel : title !== undefined ? title : route.name}
         icon={drawerIcon}
         focused={focused}
-        style={itemStyle}
+        labelStyle={[
+          fontStyles.regular,
+          { color: colors.dark, marginLeft: -20, fontWeight: 'normal' },
```
Comment (dievazqu) [inline src/navigator/DrawerNavigator.tsx:155]: Should we move this style somewhere else?

**Review by dievazqu — APPROVED** [review]
LGTM

File: src/qrcode/QRScanner.tsx (hunk 1 above)
Comment (MuckT) [reply src/qrcode/QRScanner.tsx:72]: This is there was a failing e2e test in CI where the keyboard was not dismissed. I was unable to reproduce locally and this modal is only used in e2e tests so it seems fine to have this dismiss here.
